### PR TITLE
Fix storage format v4 migration

### PR DESCRIPTION
**Pull request by turbolent**
During the last Canary spork we identified several problems with the state migration, this PR fixes those issues.

### CODE REVIEW

File: cmd/util/ledger/migrations/storage_v4.go [hunk 1]
```diff
@@ -66,36 +69,87 @@ func storageFormatV4MigrationWorker(jobs <-chan ledger.Payload, results chan<- s
...
+	// Ignore known payload keys that are not Cadence values
+
+	if bytes.HasPrefix(rawKey, publicKeyKeyPrefix) ||
+		bytes.Equal(rawKey, storageUsedKey) ||
+		bytes.Equal(rawKey, existsKey) ||
+		bytes.Equal(rawKey, frozenKey) ||
+		bytes.HasPrefix(rawKey, codeKeyPrefix) ||
+		bytes.HasPrefix(rawKey, accountAddressStateKey) ||
+		bytes.Equal(rawKey, uuidKey) ||
+		bytes.Equal(rawKey, contractNamesKey) {
```
Comment (ramtinms) [inline cmd/util/ledger/migrations/storage_v4.go:119]: While it should be the same functionality I added IsFVMStateKey to state [here](https://github.com/onflow/flow-go/pull/657/files) , so we can use it in other places and migrations.

Comment (ramtinms) [reply cmd/util/ledger/migrations/storage_v4.go:119]: or checks on the equality of owner and controller (in cases that is right) might be an extra measure to make sure we don't have any key collision.

File: cmd/util/ledger/migrations/storage_v4_test.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+	"github.com/onflow/cadence/runtime/interpreter"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/cmd/util/ledger/migrations"
+	"github.com/onflow/flow-go/engine/execution/state"
+	"github.com/onflow/flow-go/ledger"
+)
+
+type keyPair struct {
```
Comment (ramtinms) [inline cmd/util/ledger/migrations/storage_v4_test.go:14]: keyPair seems not to be in use any more

**Review by ramtinms — APPROVED** [review]
Looks good to me, just some minor suggestions

File: cmd/util/ledger/migrations/storage_v4_test.go (hunk 2 above)
Comment (turbolent) [reply cmd/util/ledger/migrations/storage_v4_test.go]: removed in 0e262d6

File: cmd/util/ledger/migrations/storage_v4.go (hunk 1 above)
Comment (turbolent) [reply cmd/util/ledger/migrations/storage_v4.go]: refactored in b4657f9

File: cmd/util/ledger/migrations/storage_v4.go [hunk 3]
```diff
@@ -66,36 +70,71 @@ func storageFormatV4MigrationWorker(jobs <-chan ledger.Payload, results chan<- s
...
 		results <- result
 	}
 }
 
+var decMode = func() cbor.DecMode {
+	decMode, err := cbor.DecOptions{
+		IntDec:           cbor.IntDecConvertNone,
+		MaxArrayElements: math.MaxInt32,
+		MaxMapPairs:      math.MaxInt32,
+		MaxNestedLevels:  256,
```
Comment (ramtinms) [inline cmd/util/ledger/migrations/storage_v4.go:87]: Are these the same params as Cadence storage?

Comment (turbolent) [reply cmd/util/ledger/migrations/storage_v4.go:87]: No, this mode uses the maximums, as we only use it to determine if some data is CBOR. The Cadence decoder then still uses it's own decoding mode to decode.

**Review by Kay-Zee — APPROVED** [review]

**Pull request merged**